File: Leetcode/122-Best-Time-to-Buy-and-Sell-Stock-II/BestTimeToBuyAndSellStockII.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        if (prices.empty()) {
            return profit;
        }
        int cost = prices[0];
        int cur = 1;
        while (cur<prices.size()) {
            if (prices[cur] <prices[cur-1]) {
                profit = profit + prices[cur-1] - cost;
                cost = prices[cur];
            }
            cur++;
        }
        profit = profit + prices[cur-1] - cost;
        return profit;
    }
};

File: Leetcode/122-Best-Time-to-Buy-and-Sell-Stock-II/BestTimeToBuyAndSellStockII.go
func maxProfit(prices []int) int {
	profit := 0
	if prices == nil || len(prices) < 2 {
		return profit
	}
	cost := prices[0]
	cur := 1
	for cur < len(prices) {
		if prices[cur] < prices[cur-1] {
			profit = profit + prices[cur-1] - cost
			cost = prices[cur]
		}
		cur++
	}
	profit = profit + prices[cur-1] - cost
	return profit
}
